### ISSUE TITLE
[ddc] staticInterop failing with The method isn't defined for the class `JavaScriptObject`

### ISSUE THREAD

**Issue by dnys1** [dart-lang/sdk#49474]
I'm experiencing an issue related to static interop in `dev` and confirmed still present in `master`. Here's the reproduction:

`interop.dart`
```dart
@JS()

import 'package:js/js.dart';
import 'package:js/js_util.dart' as js_util;

@JS()
external IndexedDB? indexedDB;

@JS()
@staticInterop
abstract class IndexedDB {}

extension IndexedDBExt on IndexedDB {
  Object open(String name) => js_util.callMethod(this, 'open', [name]);
}
```

`interop_test.dart`
```dart
import 'interop.dart';

void main() {
  indexedDB!.open('db');
}
```

Trying to build this fails with:

```console
$ dart run build_runner test -- test/interop_test.dart -p chrome
[SEVERE] build_web_compilers:ddc on test/interop_test.ddc.module:
Error compiling dartdevc module:issue|test/interop_test.sound.ddc.js

test/interop_test.dart:4:14: Error: The method 'open' isn't defined for the class 'JavaScriptObject'.
 - 'JavaScriptObject' is from 'dart:_interceptors'.
Try correcting the name to the name of an existing method, or defining a method named 'open'.
  indexedDB!.open('db');
             ^^^^
```

`dev` version:

```console
$ dart --version
Dart SDK version: 2.18.0-271.0.dev (dev) (Sat Jul 9 12:06:18 2022 -0700) on "macos_x64"
```

`master` version:

```console
$ dart --version
Dart SDK version: 2.19.0-edge.5cc2856daa9f62245ef0aaddd966321fcb0310ea (be) (Tue Jul 19 00:03:04 2022 +0000) on "macos_x64"
```

This issue is resolved by switching to `stable`.

**Comment by ykmnkmi** [comment]
Same.
```powershell
$ dart --version
Dart SDK version: 2.18.0-271.2.beta (beta) (Mon Jul 18 21:52:28 2022 +0200) on "windows_x64"
```

**Comment by srujzs** [comment]
Hi, sorry for the late response here, I've been working on a very similar bug that I hope resolves this as well. https://github.com/dart-lang/sdk/commit/459d57c8b84a6d8f7b0955326b1fdc39c6d502e2 was introduced in 2.18.0-204.0.dev which regressed some behavior around modular compilation (in an attempt to fix other issues...). The fix for this reverts the outline transformations, so my hope is once that's landed, this should be fixed. I'll respond back trying out the fix once it has landed.

**Comment by srujzs** [comment]
https://github.com/dart-lang/sdk/commit/61abaeda3f84e6d45c4a70689b08b798713bc5c1 has landed. A dev version should be available in a few hours. I believe this should resolve this issue, but if not, I'll investigate deeper. Thanks!

**Comment by dnys1** [comment]
Hi @srujzs, I was able to test on `2.19.0-52.0.dev` and can confirm the issue is resolved for me. Thanks for your help and for the great work on Dart->JS! Cheers!

**Comment by jscholler** [comment]
Hello, I just updated to stable 2.18.0 and I have a similar issue. Is that expected? Would the fix make it to 2.18.x or we should wait for 2.19.0?

**Comment by srujzs** [comment]
Yeah, I think that's expected unfortunately. The above change made it into [2.19.0-52.0.dev](https://github.com/dart-lang/sdk/releases/tag/2.19.0-52.0.dev), so it'll require using that version.
